include/fw_info.h

```diff
@@ -79,12 +103,22 @@ struct __packed fw_info {
 
 /* Static asserts to ensure compatibility */
 OFFSET_CHECK(struct fw_info, magic, 0);
-OFFSET_CHECK(struct fw_info, firmware_size, CONFIG_FW_INFO_MAGIC_LEN);
-OFFSET_CHECK(struct fw_info, firmware_version,
-	(CONFIG_FW_INFO_MAGIC_LEN + 4));
-OFFSET_CHECK(struct fw_info, firmware_address,
-	(CONFIG_FW_INFO_MAGIC_LEN + 8));
-
+OFFSET_CHECK(struct fw_info, firmware_size, 12);
+OFFSET_CHECK(struct fw_info, firmware_version, 16);
+OFFSET_CHECK(struct fw_info, firmware_address, 20);
+OFFSET_CHECK(struct fw_info, valid, 24);
+OFFSET_CHECK(struct fw_info, abi_out_len, 28);
+OFFSET_CHECK(struct fw_info, abi_out, 32);
+OFFSET_CHECK(struct fw_info, abi_in_len, 36);
+OFFSET_CHECK(struct fw_info, abi_in, 40);
+OFFSET_CHECK(struct fw_info, reserved00, 44);
+
+OFFSET_CHECK(struct fw_info_abi_request, abi_id, 0);
```

Comment:
Could these static asserts be closer to the type declaration? It comes out of order with the other type declaration first fw_info and then fw_info_abi_request

Reply:
Moved stuff around.